modules/internal/String.chpl

```diff
@@ -575,13 +578,14 @@ module String {
   }
 
   pragma "no doc"
-  proc chpl_createStringWithLiteral(x: c_string, length:int) {
+  proc chpl_createStringWithLiteral(x: c_string, length:int, numCodepoints: int) {
     // NOTE: This is a "wellknown" function used by the compiler to create
     // string literals. Inlining this creates some bloat in the AST, slowing the
     // compilation.
     return chpl_createStringWithBorrowedBufferNV(x:c_ptr(uint(8)),
                                                  length=length,
-                                                 size=length+1);
+                                                 size=length+1,
+                                                 numCodepoints);
```

Comment:
I am surprised that this works.

I think it is a very good practice to stick with named arguments once you start using them in a call. i.e. `length` and `size` are explicitly named, and so should be `numCodepoints`

Reply:
To solve this, would you say that I should have numCodepoints=cachedNumCodepoints?